CMakeLists.txt

```diff
@@ -0,0 +1,250 @@
+    option(OPENASSETIO_ENABLE_SANITIZER_UNDEFINED_BEHAVIOR
+        "Enable undefined behavior sanitizer" OFF)
+    option(OPENASSETIO_ENABLE_SANITIZER_THREAD "Enable thread sanitizer" OFF)
+    option(OPENASSETIO_ENABLE_SANITIZER_MEMORY "Enable memory sanitizer" OFF)
+endif ()
+
+
+#-----------------------------------------------------------------------
+# Enable ctest testing
+
+# To correctly generate the DartConfiguration.tcl, include the CTest
+# CMake file. By doing so 'ctest -F memcheck' works.
+# TODO(DF): ...but also creates many many pointless targets.
+#include(CTest)  # Automatically calls enable_testing()
+enable_testing()
+
+#-----------------------------------------------------------------------
+# Provides install directory variables as defined by the GNU Coding
+# Standards.
+
+include(GNUInstallDirs)
+
+#-----------------------------------------------------------------------
+# Generate OpenAssetIOConfig.cmake et al. for bundling as a CMake lib.
+
+# Targets exported to ${PROJECT_NAME}_EXPORTED_TARGETS will be available
+# for import using find_package(OpenAssetIO).
+
+include(CMakePackageConfigHelpers)
+
+set(version_config_file "${PROJECT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake")
+set(project_config_file "${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake")
+set(config_install_dir "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
+
+# Version fetched from the top level project()
+write_basic_package_version_file(
+    ${version_config_file}
+    # Version range supported only if generated by CMake >= 3.19.2.
+    COMPATIBILITY SameMinorVersion
+)
+
+configure_package_config_file(
+    cmake/Config.cmake.in
+    ${project_config_file}
+    INSTALL_DESTINATION ${config_install_dir}
+    # TODO(DF): PATH_VARS?
+)
+
+install(
+    EXPORT ${PROJECT_NAME}_EXPORTED_TARGETS
+    DESTINATION ${config_install_dir}
+    NAMESPACE ${PROJECT_NAME}::
+    FILE ${PROJECT_NAME}Targets.cmake
+)
+
+install(
+    FILES "${project_config_file}" "${version_config_file}"
+    DESTINATION "${config_install_dir}"
+)
+
+# TODO(DF): pkg-config? See OCIO.
+
+
+#-----------------------------------------------------------------------
+# Print a status dump
+
+message(STATUS "Configuring ${PROJECT_NAME} ${PROJECT_VERSION}")
+message(STATUS "CMake ${CMAKE_VERSION}")
+message(STATUS "Project build dir               = ${PROJECT_BINARY_DIR}")
+message(STATUS "Install prefix                  = ${CMAKE_INSTALL_PREFIX}")
+message(STATUS "Configuration types             = ${CMAKE_CONFIGURATION_TYPES}")
+message(STATUS "Build type                      = ${CMAKE_BUILD_TYPE}")
+message(STATUS "Build shared libs               = ${BUILD_SHARED_LIBS}")
+message(STATUS "Warnings as errors              = ${OPENASSETIO_WARNINGS_AS_ERRORS}")
+message(STATUS "Interprocedural optimization    = ${OPENASSETIO_ENABLE_IPO}")
+message(STATUS "Force position independent code = ${OPENASSETIO_ENABLE_POSITION_INDEPENDENT_CODE}")
```

Comment:
Nit: Option does not mention 'force', but status message does.

Reply:
Updated `option` description in fbea5b34

Reply:
Oh sorry, I wasn't clear.
`OPENASSETIO_ENABLE_POSITION_INDEPENDENT_CODE` doesn't say force. So I was badly hinting at renaming 'force' to 'enable'.

Reply:
Rephrased again in 310f1f34 :)